clean up gui main and fix a png.

tsteven4 opened the pull request:
- fix libpng warning: iCCP: known incorrect sRGB profile
  issue with runit.png with pngcrush.

- fix includes for gui/main.cc, including removal of
  CoreFoundation.h
- modify environmental varialble PATH  processing.
  - use toNativeSeparators.
  - use fromLocal8Bit on output of getenv.
  - use toLocal8Bit on input for putenv.
  - replace allocation of character array, conversion to
    standard string, and conversion to c_str with
    a local QByteArray and toLocal8Bit.
- use QApplication static methods instead of going directly
  to the underlying QCoreApplication static methods.

tsteven4 committed:
clean up gui main and fix a png.

- fix libpng warning: iCCP: known incorrect sRGB profile
  issue with runit.png with pngcrush.

- fix includes for gui/main.cc, including removal of
  CoreFoundation.h
- modify environmental varialble PATH  processing.
  - use toNativeSeparators.
  - use fromLocal8Bit on output of getenv.
  - use toLocal8Bit on input for putenv.
  - replace allocation of character array, conversion to
    standard string, and conversion to c_str with
    a local QByteArray and toLocal8Bit.
- use QApplication static methods instead of going directly
  to the underlying QCoreApplication static methods.

tsteven4 committed:
handle difficulties in setting path.

tsteven4 committed:
don't use methods introduced in Qt 5.10.

GPSBabelDeveloper commented on gui/main.cc:
```diff
@@ -52,15 +89,11 @@ int main(int argc, char** argv)
-  char* newPathEnv = new char[newPath.length() + 1];
-  strcpy(newPathEnv, newPath.toStdString().c_str());
-  putenv(newPathEnv);
+  set_path();
```
I wonder if we should quit trying so hard with OS specific code that's so fragile-looking. Can we look in the application dir and be done?
I think before we provided installers and encouraged people to use the GUI and the "real" app separately, we invited these issues, but it seems like anyone using the GUI that didn't do it through the installer should be able to work out broken PATH.
But I'm OK with it if it works and you're tired of thinking about this. :-)

GPSBabelDeveloper reviewed (commented):
OK

tsteven4 committed:
Revert "don't use methods introduced in Qt 5.10."

This reverts commit e6625fd2fbd66e5defaa2d8a59e6b08eaee80ee7.

tsteven4 committed:
Revert "handle difficulties in setting path."

This reverts commit f485ac0c788233fa4e4b3df3dfcb7b9253c1b9db.

tsteven4 committed:
no path setting in GUI.

instead pass the full path to gpsbabel when using QProcess::start.

in formatload.cc xlt() use qt methods to get a char* instead of
resorting to stdlib.  At this point Qt assumes all translate strings
are utf8.  However, the formatload xlt intent doesn't work because
1) MainWindow::LoadLanguage hasn't been called when
   MainWindow::loadFormats is called.
2) the translations don't exist for the format bits from gpsbabel
   that fromatload tries to translate.  I think this was the intent
   of the gpsbabel_*.ts files which are very out of date and no
   longer distributed.

GPSBabelDeveloper commented on gui/mainwindow.cc:
```diff
@@ -869,7 +895,7 @@ bool MainWindow::runGpsbabel(const QStringList& args, QString& errorString,
   QProcess* proc = new QProcess(nullptr);
   QString name = "gpsbabel";
-  proc->start(name, args);
+  proc->start(QApplication::applicationDirPath() + '/' + name, args);
```
Consider refactoring this and the one above in 505 to use a simple one-liner that builds the full path to the executable in a single place before commit.

GPSBabelDeveloper reviewed (commented):
This eliminates lots of icky stuff. I like it. Please consider making it a single place where we find our own executable, but LGTM. I did verify on MacOS that gpsbabel and gpsbabelfe were siblings in GPSBabelFE.app/Contents/MacOS and that it wasn't buried in resources/
